server/chat-plugins/room-events.ts

```diff
@@ -19,30 +43,57 @@ function formatEvent(event: RoomEvent, showAliases?: boolean) {
 	if (!isNaN(timeRemaining)) {
 		explanation = `This event will start in: ${Chat.toDurationString(timeRemaining, {precision: 2})}`;
 	}
+
+	const eventID = toID(event.eventName);
+	const aliases = getAliases(room, eventID);
+	const categories = getAllCategories(room).filter(
+		category => (room.settings.events![category] as RoomEventCategory).events.includes(eventID)
+	);
+
 	let ret = `<tr title="${explanation}">`;
 	ret += Utils.html`<td>${event.eventName}</td>`;
-	ret += showAliases ? Utils.html`<td>${event.aliases?.join(", ")}</td>` : ``;
+	if (showAliases) ret += Utils.html`<td>${aliases.join(", ")}</td>`;
+	if (showCategories) ret += Utils.html`<td>${categories.join(", ")}</td>`;
 	ret += `<td>${Chat.formatText(event.desc, true)}</td>`;
 	ret += Utils.html`<td><time>${event.date}</time></td></tr>`;
 	return ret;
 }
 
-function getAllAliases(room: Room) {
+function getAliases(room: Room, eventID?: ID) {
 	if (!room.settings.events) return [];
 	const aliases: string[] = [];
-	for (const event of Object.values(room.settings.events)) {
-		if (event.aliases) aliases.push(...event.aliases);
+	for (const aliasID in room.settings.events) {
+		if (
+			'eventID' in room.settings.events[aliasID] &&
+			(!eventID || (room.settings.events[aliasID] as RoomEventAlias).eventID === eventID)
```

Comment:
```suggestion
			(!eventID || room.settings.events[aliasID].eventID === eventID)
```

I'm surprised this isn't recognized as a useless cast.